zerver/lib/upload.py

```diff
@@ -89,6 +89,7 @@ def sanitize_name(value: str) -> str:
     modified by:
     * adding '.' to the list of allowed characters.
     * preserving the case of the value.
+    * not stripping trailing dashes and underscores.
```

Comment:
Alternatively, we could change this behavior to be like slugify. Not sure what's better for this use case (if we port the change from https://github.com/django/django/commit/0382ecfe020b4c51b4c01e4e9a21892771e66941 we would be stripping dashes and underscore, but not dots which is a bit inconsistent - or we could strip dots too, but then that requires documenting too as a difference. overall seems simplest to do it like in this commit i think...)

Reply:
I think it'd probably be reasonable to make the trailing dashes/underscores like slugify.